### PR TITLE
Add a retry to remove the vttablet directory during upgrade/downgrade backup tests

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR fixes a small issue during the manual upgrade/downgrade backup tests. The issue happens when for some reason the deletion of the VTTablet directory fails due to used files, leading to an error code `39` from `rm -Rf`. This issue can be seen as follows in the logs of the `Stop tablets` step:

```
Shutting down tablet zone1-301
Stopping vttablet...
Shutting down mysql zone1-301
Shutting down MySQL for tablet zone1-0000000301...
Removing tablet directory zone1-301
Successfully deleted 1 tablets
rm: cannot remove '/tmp//vt_0000000301': Directory not empty
```

This silent error leads to the failure of step `Start new tablet and restore`. Since the VTTablet's directory was partially removed, VTTablet thinks it has to start from an existing data dir (which is not what we want). And because the VTTablet directory has been mostly emptied already, the `my.cnf` file does not exist, leading to a fatal error (also visible in the logs):

```
Starting MySQL for tablet zone1-0000000301...
Resuming from existing vttablet dir:
    /tmp//vt_0000000301
E1208 15:33:57.111914   44151 mysqlctl.go:276] failed to find mysql config: couldn't read my.cnf file: open /tmp/vt_0000000301/my.cnf: no such file or directory
```

After some local tests, it seems like adding a sleep before the `rm -Rf` is sufficient. I then decided to add a 30 second-long retry loop where we attempt to remove the VTTablet directory every second.

Moreover, the list of files to watch in the backups manual workflows have changed to include the `examples/**` path as we use it in our tests.

### CODE REVIEW

File: examples/backups/stop_tablets.sh [hunk 1]
```diff
@@ -30,7 +30,25 @@ for tablet in 100 200 300; do
...
-      rm -Rf $VTDATAROOT/vt_0000000$uid
+
+      for ((i=0; i<30; i++)); do
+          # Redirect stderr to a temporary file
+          temp_file=$(mktemp)
+          rm -Rf $VTDATAROOT/vt_0000000$uid 2>$temp_file
+
+          if grep -q 'Directory not empty' $temp_file; then
+              echo "Directory not empty, retrying..."
+          elif [ ! -s $temp_file ]; then
```
Comment (mattlord) [inline examples/backups/stop_tablets.sh:41]: We should quote the `$temp_file` variable everywhere to be safe (can run shellcheck on the file too).

**Review by mattlord — APPROVED** [review]
LGTM! Only had a small nit about quoting the variable when used (it could potentially have whitespace). FWIW, I also think it's more descriptive to say that we're adding a retry rather than a timeout.

I remember running into this same `Directory not empty` error when dealing some other backup related test flakiness some time ago. I think it might have been here: https://github.com/vitessio/vitess/pull/11352/files#diff-4f8ae786620a21df8ced1435fc0318b8abba570dfea4c2177b1ecb725b8c5073L236-R244

Thank you for working on this!

**Comment by frouioui** [comment]
> Only had a small nit about quoting the variable when used (it could potentially have whitespace).

I will go ahead and fix it, thanks!

> I also think it's more descriptive to say that we're adding a retry rather than a timeout

Got it! I modified both title and description.


> I remember running into this same `Directory not empty` error when dealing some other backup related test flakiness some time ago. I think it might have been here: [#11352 (files)](https://github.com/vitessio/vitess/pull/11352/files#diff-4f8ae786620a21df8ced1435fc0318b8abba570dfea4c2177b1ecb725b8c5073L236-R244)

Nice, thank you! This is useful.

File: examples/backups/stop_tablets.sh (hunk 1 above)
Comment (frouioui) [reply examples/backups/stop_tablets.sh]: Addressed via 8841067156

**Review by rohit-nayak-ps — APPROVED** [review]

**Pull request merged**